asserter/account.go

```diff
@@ -35,17 +35,16 @@ func ContainsCurrency(currencies []*types.Currency, currency *types.Currency) bo
 	return false
 }
 
-// assertBalanceAmounts returns an error if a slice
-// of types.Amount returned as an types.AccountIdentifier's
-// balance is invalid. It is considered invalid if the same
+// assertUniqueAmounts returns an error if a slice
+// of types.Amount is invalid. It is considered invalid if the same
 // currency is returned multiple times (these shoould be
 // consolidated) or if a types.Amount is considered invalid.
-func assertBalanceAmounts(amounts []*types.Amount) error {
+func assertUniqueAmounts(amounts []*types.Amount) error {
 	currencies := make([]*types.Currency, 0)
 	for _, amount := range amounts {
-		// Ensure a currency is used at most once in balance.Amounts
+		// Ensure a currency is used at most once
 		if ContainsCurrency(currencies, amount.Currency) {
-			return fmt.Errorf("currency %+v used in balance multiple times", amount.Currency)
+			return fmt.Errorf("currency %+v used multiple times", amount.Currency)
 		}
 		currencies = append(currencies, amount.Currency)
```

Comment:
The complexity of `assertUniqueAmounts` is `O(n^2)`. We should be able to do `O(n)`

Reply:
That's a good call out. This is some old code I haven't touched in a while.

Reply:
Usually this array is at most 1 or 2 items but could potentially be much larger when implementations support multiple tokens.

We can punt on this until then!